mlrun/package/context_handler.py

```diff
@@ -187,61 +187,43 @@ def log_outputs(
         Log the given outputs as artifacts (or results) with the stored context. Errors raised during the packing will
         be ignored to not fail a run. A warning with the error wil be printed.
 
+        Only the logging worker will pack and log the outputs.
+
         :param outputs:   List of outputs to log.
         :param log_hints: List of log hints (logging configurations) to use.
         """
-        # Verify the outputs and log hints are the same length:
-        if len(outputs) != len(log_hints):
-            self._context.logger.warn(
-                f"The amount of outputs objects returned from the function ({len(outputs)}) does not match the amount "
-                f"of provided log hints ({len(log_hints)})."
+        # Pack and log only from the logging worker (in case of multi-workers job like OpenMPI):
+        if self._is_logging_worker():
```

Comment:
I suggest to check if not is logging worker - log skipping and return.
No need for else clause and is more readable

Reply:
I know, but I am clearing the packagers garbage at the end of the run regardless of logging outputs. Outputs can be produced when parsing inputs as well. For example, reading an `xml` file, downloading it to temp, cast to `DataFrame`. At the end of the run I wish this file I downloaded to be deleted.